### PR TITLE
Enable listening to Microsoft.Data.SqlClient.EventSource on .NET full framework

### PR DESCRIPTION
**Pull request by Mpdreamz**
This should fix https://github.com/elastic/apm-agent-dotnet/issues/2038

Still working on reproducing this.

### CODE REVIEW

**Review by gregkalapos — APPROVED** [review]

**Comment by ddgpablo** [comment]
Hello,

To reproduce this issue you just need to create a .NETFramework 4.7.2 web app and perform a SQL operation when hitting one of the endpoints. 

I tested this fix and it's still not creating the spans due to these checks:

https://github.com/elastic/apm-agent-dotnet/blob/20fea7888fd3a044e13fb34c3e657d26502a6911/src/Elastic.Apm.SqlClient/SqlEventListener.cs#L75

https://github.com/elastic/apm-agent-dotnet/blob/20fea7888fd3a044e13fb34c3e657d26502a6911/src/Elastic.Apm.SqlClient/SqlEventListener.cs#L121

The payload looks to be always 5 in `ProcessBeginExecute` and 4 in `ProcessEndExecute`. Is this specific for .NET Framework? Should we alter the count check?

Kind regards,
Pablo

**Comment by Mpdreamz** [comment]
> The payload looks to be always 5 in ProcessBeginExecute and 4 in ProcessEndExecute. Is this specific for .NET Framework? Should we alter the count check?

Thanks for reporting this @ddgpablo found the same while working on a reproduction. 
This took a bit longer because I was fighting: https://github.com/dotnet/SqlClient/issues/801 which was caused because I was refencing `net461` instead of `net462`. 

Our payload checks are too stringent, `SqlClientEventSource` reserves the first sequence of the payload but MAY add more tokens to the payload in the future see:

https://github.com/dotnet/SqlClient/blob/3a41288f2b67307a1f816761deb73785247c85c9/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlClientEventSource.cs#L1009

**Comment by ddgpablo** [comment]
Many thanks for your answer @Mpdreamz. Is the PR ready to review? Let me know and I can test it again.

**Comment by Mpdreamz** [comment]
@ddgpablo it is!

**Comment by ddgpablo** [comment]
Great, that works! Many thanks for your help mate.

File: sample/Sample.Microsoft.Data.SqlClient/Program.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+Console.WriteLine("Performing database calls");
+
+var transaction = Agent.Tracer.StartTransaction("test-sql-client-calls", "app");
+DoSqlClientCall(connectionString);
+transaction.End();
+
+Console.WriteLine("Done!");
+
+static void DoSqlClientCall(string connectionString)
```
Comment (gregkalapos) [inline sample/Sample.Microsoft.Data.SqlClient/Program.cs:26]: Could we move this into a test? From what I understand, we do not run this currently anywhere. I think `Elastic.Apm.SqlClient.Tests` would be a better place for this.

**Review by gregkalapos — APPROVED** [review]
LGTM, just one thing about testing.

**Pull request merged**